derrida/interventions/management/commands/insertion_data.py

```diff
@@ -0,0 +1,145 @@
+'''
+Manage command to export intervention insertions data for use by others.
+
+Generates a CSV and JSON file with details for all insertions
+documented via IIIF canvas image labels in the database.
+
+Takes an optional argument to specify the output directory. Otherwise,
+files are created in the current directory.
+'''
+
+
+import codecs
+from collections import OrderedDict, defaultdict
+import csv
+import json
+import os.path
+import re
+from itertools import groupby
+
+from django.db.models import ObjectDoesNotExist
+from djiffy.models import Canvas
+
+from derrida.books.management.commands import reference_data
+from derrida.interventions.models import Intervention
+
+
+# regex to extract insertion label common to all images
+# for a single insertion from the canvas label
+# grouping for full label, page range/label in book, insertion page label
+# FIXME: negative lookahead for with in page number is not working properly
+RE_INSERTION_LABEL = re.compile(r'(?P<label>(?P<page>.*(?! with))( with)?Insertions? [A-Z])(?P<insertion_page>.*$)')
```

Comment:
I won't fully understand how you're parsing things here without going through those notes myself. I'm skipping this for now, but please let me know if you'd like me to check your work more closely!